init-tools.sh

```diff
@@ -98,15 +117,21 @@ if [ ! -e $__DOTNET_PATH ]; then
 
     echo "Installing dotnet cli..."
     __DOTNET_LOCATION="https://dotnetcli.azureedge.net/dotnet/Sdk/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.tar.gz"
-    # curl has HTTPS CA trust-issues less often than wget, so lets try that first.
-    echo "Installing '${__DOTNET_LOCATION}' to '$__DOTNET_PATH/dotnet.tar'" >> $__init_tools_log
-    if command -v curl > /dev/null; then
-        curl --retry 10 -sSL --create-dirs -o $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
-    else
-        wget -q -O $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
-    fi
-    cd $__DOTNET_PATH
-    tar -xf $__DOTNET_PATH/dotnet.tar
+    retries=5
+    waitFactor=6 
+    installDotNetCLI() {
+        echo "Installing '${__DOTNET_LOCATION}' to '$__DOTNET_PATH/dotnet.tar'" >> $__init_tools_log
+        rm -rf -- $__DOTNET_PATH/*
```

Comment:
Quotes for `"$__DOTNET_PATH/*"` would be good just in case a space makes it here, even though the rest of the script isn't consistent about it (yet).

Reply:
I think that would also go for __init_tools_log too

Reply:
Yeah, I focused on this line because it's new, but basically any line where a variable is being passed would be improved by adding quotes.

Reply:
why are you removing * as opposed to just dotnet.tar?

Reply:
I'm removing * because of the slim chance that tar may extract something and then fail.  I want to get the directory back to the initial state, which is empty.